### DISCUSSION TITLE
Saving Customer Payment Source

### DISCUSSION THREAD

**Discussion by brookecm28** [digital-pros/commerce-authorize#37] [Q&A]
Good morning @benlobach :)

I have another question for you. I was able to get the payments connected through a little more custom coding. However, when I'm having trouble getting the customerProfileId & customerPaymentProfileId when a payment is saved in authorize.net.

Is this getting returned somewhere in the response from craft's standard Pay action?

**Comment by benlobach** [comment]
@brookecm28 - we have a function in our plugin that saves these values into the database, and then retrieves them as needed, but it wouldn't be part of the standard pay action.

**Reply by brookecm28** [reply]
Ok, so once a payment source has been created in the CIM (which is happening), how would I go about retrieving that payment source to be able to use it again?

Is this the `populateFromPaymentSource` function?

**Reply by benlobach** [reply]
It's saved as the token on the Payment Source within the Commerce Authorize plugin. If you can recall the payment source, or query the database, you should be able to retrieve the values you need.

https://github.com/digital-pros/commerce-authorize/blob/f7e5dc31121d40455451f0a9dae172cfcf95a5b6/src/gateways/Subscriptions.php#L378
